deploy/helm/kubecf/assets/scripts/jobs/pxc/migrate_charset.sh

```diff
@@ -5,41 +5,100 @@
 
 set -o errexit -o nounset -o pipefail
 
+cat > "${HOME}/.my.cnf" <<EOF
+[mysql]
+host=${DATABASE_HOST}
+user=root
+password=${DATABASE_ROOT_PASSWORD}
+EOF
+
 echo "Waiting for database to be ready..."
-until echo "SELECT 'Ready!'" | mysql --host="${DATABASE_HOST}" --user=root --password="${DATABASE_ROOT_PASSWORD}" --connect-timeout=3 2> /dev/null; do
+until echo "SELECT 'Ready!'" | mysql --connect-timeout=3 1> /dev/null 2> /dev/null; do
   sleep 1
 done
 
-function alter_database() {
+function show_tables() {
```

Comment:
Wouldn't it be possible to use SQL queries to obtain the info from the tables instead of show tables and awk?
Something like:
```
USE information_schema;

SELECT concat("ALTER TABLE `",table_schema,"`.`",table_name, "` CHANGE `",column_name,"` `",column_name,"` ",data_type,"(",character_maximum_length,")
                  CHARACTER SET utf8 COLLATE utf8_unicode_ci",IF(is_nullable="YES"," NULL"," NOT NULL"),";") AS _sqlquery
FROM `COLUMNS` WHERE table_schema LIKE "${database}" AND data_type IN ('varchar','char');

SELECT concat("ALTER TABLE `",table_schema,"`.`",table_name, "` CHANGE `",column_name,"` `",column_name,"` ",data_type," CHARACTER SET utf8 COLLATE utf8_unicode_ci",IF(is_nullable="YES"," NULL"," NOT NULL"),";") as _sqlquery
FROM `COLUMNS` WHERE table_schema LIKE "${database}" AND data_type IN ('text','tinytext','mediumtext','longtext');
```

Reply:
I'd love to take this suggestion now, but this would involve a lot of testing again. `awk` was my friend to experiment a lot and compare the dumps from pre-migration and post-migration.